### PR TITLE
feat: allow setting notification read also from web notifications 

### PR DESCRIPTION
**Pull request by drodil**
## Hey, I just made a Pull Request!

Allow setting notification read also from web notifications and snackbar when opening link similarly like in the #24508 for the notifications table.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by drodil** [comment]
Not stale, need a rebase

**Comment by drodil** [comment]
Not stale, needs review, @mareklibra if you have time :)

File: plugins/notifications/api-report.md
Comment (Rugvip) [inline plugins/notifications/api-report.md:186]: Hmm, how come this is exported in the first place, should we keep it internal? A think like this should otherwise live in `notifications-react`. If we want to keep it exported, let's make it an options object instead: `useWebNotifications(options: {enabled?: boolean, markAsReadOnLinkOpen?: boolean})`

**Review by Rugvip — CHANGES_REQUESTED** [review]

File: plugins/notifications/api-report.md
Comment (drodil) [reply plugins/notifications/api-report.md:186]: Yeah, it probably should be internal, I will change!

File: .changeset/ten-penguins-roll.md
Comment (Rugvip) [inline .changeset/ten-penguins-roll.md:2]: Now that we're making some breaking change this should be `minor`. I think immediate breakage is fine though

File: plugins/notifications/api-report.md
Comment (Rugvip) [inline plugins/notifications/api-report.md:106]: I wonder if there's really much variance here across apps? I.e. either it should be the built-in behavior, or it might have some problems that we should solve regardless. I'm thinking we should aim to not need to have this configurable

**Review by Rugvip — CHANGES_REQUESTED** [review]

File: plugins/notifications/api-report.md
Comment (drodil) [reply plugins/notifications/api-report.md:106]: I would vote for making this the default behavior but not sure what for example @mareklibra thinks?

File: plugins/notifications/src/components/NotificationsSideBarItem/NotificationsSideBarItem.tsx
Comment (mareklibra) [inline plugins/notifications/src/components/NotificationsSideBarItem/NotificationsSideBarItem.tsx:278]: This is a good change to make it working in all (?) browsers.
On the other hand, the browser notifications will be shown after visiting the page only. This can be confusing.

What about keeping that for the sake of compatibility here and having the former `useEffect()` as well? If the browser does not enforce an user action to check permission, the user experience will be better. The only drawback is a `[Violation] ...` message in the log without consequences.

File: plugins/notifications/api-report.md
Comment (mareklibra) [reply plugins/notifications/api-report.md]: I don't think we need this to be configurable.

File: plugins/notifications/src/components/NotificationsSideBarItem/NotificationsSideBarItem.tsx
Comment (drodil) [reply plugins/notifications/src/components/NotificationsSideBarItem/NotificationsSideBarItem.tsx:278]: I think this is more future-proof if they decide that it is required to have the user interaction and will not work the other way at all. I don't think it's confusing as it will ask the permission once you go to the notifications page the first time and never after that. Other option would be to have some sort of banner/modal to ask for user permission on page load but that requires more work that can be separated to own PR if we decide to do that. 

What do you think?

Comment (mareklibra) [reply plugins/notifications/src/components/NotificationsSideBarItem/NotificationsSideBarItem.tsx:278]: I am for the most intrusive approach as possible, so no banners, etc. :-)

If we keep recent implementation, we will get uniform behavior across browsers, which is probably better. So let's keep it as it is.

**Review by mareklibra — APPROVED** [review]

**Comment by drodil** [comment]
@Rugvip mind re-checking this? Thanks!

**Comment by drodil** [comment]
Not stale, waiting for final review, ping @Rugvip

**Review by mareklibra — APPROVED** [review]

**Comment by drodil** [comment]
Not stale, waiting for merge, ping @Rugvip

**Review by Rugvip — APPROVED** [review]
Sorry, didn't realize I had blocked this one. Back now though 😅, :shipit:

**Comment by drodil** [comment]
Can we get this merged? Ping @Rugvip / @mareklibra

**Pull request merged**
